packages/aws-cdk-lib/custom-resources/lib/provider-framework/runtime/framework.ts

```diff
@@ -50,8 +50,6 @@ async function onEvent(cfnRequest: AWSLambda.CloudFormationCustomResourceEvent)
     input: JSON.stringify(resourceEvent),
   };
 
-  log('starting waiter', waiter);
```

Comment:
Maybe some log line that still has waiter info as a stop gap.
Like `log('starting waiter', { stateMachineArn: waiter.stateMachineArn, name: waiter.name);`?

Reply:
Yes, sounds good since the issue is mainly from the `resourceEvent`.